lib/workers/repository/update/branch/get-updated.ts

```diff
@@ -305,6 +306,19 @@ export async function getUpdatedPackageFiles(
           updatedFileContents[packageFile] = newContent;
           delete nonUpdatedFileContents[packageFile];
         }
+        if (
+          manager === 'nix' &&
+          upgrade.currentValue === upgrade.newValue &&
+          upgrade.currentDigest &&
+          upgrade.newDigest &&
+          upgrade.currentDigest !== upgrade.newDigest
+        ) {
+          logger.debug(
+            { packageFile, depName },
+            'Nix digest-only update - tracking for artifact update',
+          );
+          artifactUpdateNeeded[packageFile] = newContent;
+        }
```

Comment:
no, we don't want special handling here. It must be handled in manager.

Reply:
I don't think this can be done in the manager with the current model. If there are no changes to flake.nix then Renovate won't run updateArtifacts. At least, I couldn't figure out how to do this work.

Reply:
@viceice Let me know how you want to proceed here